python/src/spark_rapids_ml/dbscan.py

```diff
@@ -0,0 +1,747 @@
+    """
+    The Density-Based Spatial Clustering of Applications with Noise (DBSCAN) is a non-parametric
+    data clustering algorithm based on data density. It groups points close to each other that form a dense cluster
+    and mark the far-away points as noise and exclude them from all clusters.
+
+    Parameters
+    ----------
+    featuresCol: str or List[str]
+        The feature column names, spark-rapids-ml supports vector, array and columnar as the input.\n
+            * When the value is a string, the feature columns must be assembled into 1 column with vector or array type.
+            * When the value is a list of strings, the feature columns must be numeric types.
+
+    predictionCol: str
+        the name of the column that stores cluster indices of input vectors. predictionCol should be set when users expect to apply the transform function of a learned model.
+
+    num_workers:
+        Number of cuML workers, where each cuML worker corresponds to one Spark task
+        running on one GPU. If not set, spark-rapids-ml tries to infer the number of
+        cuML workers (i.e. GPUs in cluster) from the Spark environment.
+
+    eps: float (default = 0.5)
+        The maximum distance between 2 points such they reside in the same neighborhood.
+
+    min_samples: int (default = 5)
+        The number of samples in a neighborhood such that this group can be considered as
+        an important core point (including the point itself).
+
+    metric: {'euclidean', 'cosine', 'precomputed'}, default = 'euclidean'
+        The metric to use when calculating distances between points.
+        If metric is 'precomputed', X is assumed to be a distance matrix and must be square.
+        The input will be modified temporarily when cosine distance is used and the restored input matrix might not match completely due to numerical rounding.
+
+    verbose: int or boolean (default=False)
+        Logging level.
+            * ``0`` - Disables all log messages.
+            * ``1`` - Enables only critical messages.
+            * ``2`` - Enables all messages up to and including errors.
+            * ``3`` - Enables all messages up to and including warnings.
+            * ``4 or False`` - Enables all messages up to and including information messages.
+            * ``5 or True`` - Enables all messages up to and including debug messages.
+            * ``6`` - Enables all messages up to and including trace messages.
+
+    max_mbytes_per_batch(optional): int
+        Calculate batch size using no more than this number of megabytes for the pairwise distance computation.
+        This enables the trade-off between runtime and memory usage for making the N^2 pairwise distance computations more tractable for large numbers of samples.
+        If you are experiencing out of memory errors when running DBSCAN, you can set this value based on the memory size of your device.
+
+    calc_core_sample_indices(optional): boolean (default = True)
```

Comment:
If true, do we ever use the computed core indices? Seems computed only but never used.

Reply:
Yeah, but the default from cuML is True so I think this should be a match

Reply:
Ok to diverge from cuml if no advantage to expose (say it is used internally to cuml testint).